deepmd/utils/data.py

```
@@ -707,6 +798,140 @@
+        # Branch 2: File exists, use memmap
+        mmap_obj = self._get_memmap(path)
+        # Slice the single frame and make an in-memory copy for modification
+        if mmap_obj.ndim == 0:
+            # Scalar array
+            data = mmap_obj.copy().astype(dtype, copy=False)
+        elif mmap_obj.ndim == 1:
+            # Single-frame file (shape: [ndof]); only frame_idx==0 is valid
+            if frame_idx != 0:
+                raise IndexError(
+                    f"frame index {frame_idx} out of range for single-frame file: {path}"
+                )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)